exonum/src/storage/db.rs

```diff
@@ -34,9 +38,43 @@ impl Patch {
         Self { changes: HashMap::new() }
     }
 
-    /// Returns amount of changes.
+    /// Returns changes for the given name.
+    fn get_changes(&self, name: &str) -> Option<&Changes> {
+        self.changes.get(name)
+    }
+
+    /// Returns a mutable reference to the changes corresponding to the `name`.
+    fn get_changes_mut(&mut self, name: &str) -> Option<&mut Changes> {
+        self.changes.get_mut(name)
+    }
+
+    /// Creates a changes map for the given name if doesn't exist already.
+    fn create_changes_if_needed(&mut self, name: &str) {
+        if !self.changes.contains_key(name) {
+            self.changes.insert(name.to_string(), BTreeMap::new());
+        }
+    }
+
+    /// Inserts changes with the given name.
+    fn insert_changes(&mut self, name: String, changes: Changes) {
+        self.changes.insert(name, changes);
+    }
+
+    pub(crate) fn into_iter(self) -> HashMapIntoIter<String, Changes> {
+        self.changes.into_iter()
+    }
+
+    pub(crate) fn iter(&self) -> HashMapIter<String, Changes> {
+        self.changes.iter()
+    }
+
+    /// Returns the number of changes.
     pub fn len(&self) -> usize {
-        self.changes.len()
+        let mut count = 0;
+        for cf in self.iter() {
+            count += cf.1.len();
+        }
+        count
```

Comment:
May be something like this will be better ?
`self.iter().fold(0, |acc, &cf| acc + cf.1.len())`

Reply:
https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.sum

Reply:
`sum` sums the *elements* of an iterator, so it cannot be used here. :disappointed: